hw/ip/otbn/util/shared/operand.py

```diff
@@ -309,47 +312,56 @@ def op_val_to_enc_val(self,
         if self.width is None:
             return None
 
+        # The operand can have two offsets applied to it. The PC offset is
+        # applied before shifting, the fixed_offset is applied after shifting.
```

Comment:
I'm not sure that this comment really helps (or, at least, it rather confused me). In particular, this code wasn't "applying an offset": it was converting the operand value into an offset (from the PC). Maybe remove this comment and rename `offset_val` to `pc_relative_val` (which gets rid of the confusing "offset" name entirely).